src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl_ModelComputation.cs

```diff
@@ -54,15 +56,26 @@ private void EnqueueComputeDataModelTask()
             if (filePath is null)
                 return null;
 
+            // If we are creating the document symbol tree for the first time, show the progress bar to the user.
+            if (!SymbolTreeIsInitialized)
+                UpdateProgressBarVisibility(Visibility.Visible);
+
             // Ensure we switch to the threadpool before calling ComputeModelAsync. It ensures that fetching and processing the document
             // symbol data model is not done on the UI thread.
             await TaskScheduler.Default;
 
             var model = await ComputeModelAsync().ConfigureAwait(false);
 
             // The model can be null if the LSP document symbol request returns a null response.
-            if (model is not null)
-                EnqueueFilterAndSortDataModelTask();
+            // In this case, we want to hide the progress bar since there is nothing to show.
+            if (model is null)
```

Comment:
note: i think the use of a 'null' model is really adding complexity to your code.  My recommendation would be to always do all steps of your queues regardless of whether or not hte model is null or not.  This means you'll always end up performing all steps and you won't have to worry about being in an an inconsistent state if you do an early bail out.